src/app/general/pages/gen-profile/gen-profile.component.ts

```diff
@@ -1,143 +1,155 @@
-      borderColor: 'rgba(148,159,177,1)',
-      pointBackgroundColor: 'rgba(148,159,177,1)',
-      pointBorderColor: '#fff',
-      pointHoverBackgroundColor: '#fff',
-      pointHoverBorderColor: 'rgba(148,159,177,0.8)'
-    },
-    { // dark grey
-      backgroundColor: 'rgba(77,83,96,0.2)',
-      borderColor: 'rgba(77,83,96,1)',
-      pointBackgroundColor: 'rgba(77,83,96,1)',
-      pointBorderColor: '#fff',
-      pointHoverBackgroundColor: '#fff',
-      pointHoverBorderColor: 'rgba(77,83,96,1)'
-    },
-    { // grey
-      backgroundColor: 'rgba(148,159,177,0.2)',
-      borderColor: 'rgba(148,159,177,1)',
-      pointBackgroundColor: 'rgba(148,159,177,1)',
-      pointBorderColor: '#fff',
-      pointHoverBackgroundColor: '#fff',
-      pointHoverBorderColor: 'rgba(148,159,177,0.8)'
-    },
-    { // grey
-      backgroundColor: 'rgba(148,159,177,0.2)',
-      borderColor: 'rgba(148,159,177,1)',
-      pointBackgroundColor: 'rgba(148,159,177,1)',
-      pointBorderColor: '#fff',
-      pointHoverBackgroundColor: '#fff',
-      pointHoverBorderColor: 'rgba(148,159,177,0.8)'
+
+    ngOnInit() {
+        this.pageService.isProfilePage(true);
+        this.getUser();
+        this.getGrades();
+        this.setPerformanceGraph();
     }
-  ];
-
-  public lineChartLegend: boolean = true;
-  public lineChartType: string = 'line';
-
-  /**
-   * @constructor
-   * 
-   * @param userService uses the UserService to obtains data needed for user
-   */
-  constructor(
-    private pageService: PageService,
-    private userService: UserService
-  ) {
-  }
-
-  ngOnInit() {
-    this.getUser();
-    this.getGrades();
-    this.pageService.isProfilePage(true);
-  }
-
-
-  /**
-   * @public
-   * 
-   * @summary Triggers when chart is clicked
-   * @param e 
-   */
-  public chartClicked(e: any): void {
-    console.log(e);
-  }
-
-  /**
-   * @public
-   * 
-   * @summary Triggers when chart is hovered
-   * @param e 
-   */
-  public chartHovered(e: any): void {
-    console.log(e);
-  }
-
-  /**
-   * @summary Stores the user's grades
-   */
-  getGrades(): void {
-    this.grades = TOTXP ? TOTXP : [];
-    let dataGrade: number[] = [];
-    let max: number = MAXXP ? MAXXP : 10;
-    for (let i = 0; i < this.grades.length; i++) {
-      let perc: number = this.grades[i] / MAXXP * 100;
-      dataGrade.push(Math.round((perc + 0.00001) * 100) / 100);
+
+    /**
+     * Stores the current user
+     * @description Stores the current user to the 'user' variable from the user.service function
+     */
+    getUser(): void {
+        this.userService.getUser("1")
+            .subscribe(user => this.user = user);
     }
-    let dataLine: any = {
-      data: dataGrade,
-      label: 'CMSC 128'
-    };
-    this.lineChartData.push(dataLine);
-  }
 
-  getUser(): void {
-    this.userService.getUser("1")
-      .subscribe(user => this.user = user);
-  }
+    /**
+     * Stores the user's grades
+     */
+    getGrades(): void {

```

Comment:
add opinion as comments of the function as to how these grades be fetched from the server, or expected data structure of it.